etc/spack/defaults/modules.yaml

```diff
@@ -25,16 +25,6 @@ modules:
       - MANPATH
     share/aclocal:
       - ACLOCAL_PATH
-    lib:
-      - LIBRARY_PATH
-    lib64:
-      - LIBRARY_PATH
-    include:
-      - C_INCLUDE_PATH
-      - CPLUS_INCLUDE_PATH
-      # The INCLUDE env variable specifies paths to look for
-      # .mod file for Intel Fortran compilers
-      - INCLUDE
```

Comment:
Should `INCLUDE` be kept for Fortran?

Reply:
it's Windows specific as far as I know.

Reply:
I quickly read through #19840 and #14749 because I remember this came up before. The Windows specificity was missed during the `CPATH` change but if I am understanding correctly, which may not be the case, `INCLUDE` is used for finding gfortran `.mod` files. If that is the case, then dropping `INCLUDE` would break gfortran module finding much like dropping `CPATH` broke it for `ifort`.

Reply:
I think gfortran uses `INCLUDE`, the documentation says it uses the same environment variables as gcc and I couldn't find `INCLUDE` in the list.

Reply:
> it's Windows specific as far as I know.

P.S. Windows support is coming soon! So we should keep this even if it's only used on Windows.

Reply:
I think this is the relevant text from the gfortran man page:
```
-Idir
           These affect interpretation of the "INCLUDE" directive (as well as of the "#include" directive of the cpp
           preprocessor).

           Also note that the general behavior of -I and "INCLUDE" is pretty much the same as of -I with "#include"
           in the cpp preprocessor, with regard to looking for header.gcc files and other such things.

           This path is also used to search for .mod files when previously compiled modules are required by a "USE"
           statement.
```

Reply:
> These affect interpretation of the "INCLUDE" directive

I think they mean the Fortran `INCLUDE` directive (https://docs.oracle.com/cd/E19957-01/805-4939/6j4m0vna1/index.html).

Reply:
Yeah, that part applies to the `include` directive, not the environment variable. `gfortran` man pages do not mention an `INCLUDE` environment variable. `ifort 19.1` man pages only mention `CPATH (Linux* and macOS*)` and `GXX_INCLUDE (Linux*)` but INCLUDE is never mentioned as a variable that's used even though others for Windows are listed. And if this really was used on some older version of ifort on Windows, then for consistency across platforms we should probably drop it too?